rascaline/src/calculators/composition.rs

```diff
@@ -0,0 +1,169 @@
+        return "atom centered composition features".into();
+    }
+
+    fn parameters(&self) -> String {
+        return serde_json::to_string(self).expect("failed to serialize to JSON");
+    }
+
+    fn keys(&self, systems: &mut [Box<dyn System>]) -> Result<Labels, Error> {
+        return CenterSpeciesKeys.keys(systems);
+    }
+
+    fn samples_names(&self) -> Vec<&str> {
+        return SamplesPerAtom::samples_names();
+    }
+
+    fn samples(&self, keys: &Labels, systems: &mut [Box<dyn System>]) -> Result<Vec<Arc<Labels>>, Error> {
+        assert_eq!(keys.names(), ["species_center"]);
+        let mut samples = Vec::new();
+        for [species_center] in keys.iter_fixed_size() {
+            let builder = SamplesPerAtom {
+                species_center: SpeciesFilter::Single(species_center.i32())
+            };
+    
+            samples.push(builder.samples(systems)?);
+        }
+
+        return Ok(samples);
+    }
+
+    fn supports_gradient(&self, parameter: &str) -> bool {
+        match parameter {
+            "positions" => true,
+            "cell" => true,
+            _ => false,
+        }
+    }
+
+    fn positions_gradient_samples(&self, keys: &Labels, samples: &[Arc<Labels>], systems: &mut [Box<dyn System>]) -> Result<Vec<Arc<Labels>>, Error> {
+        debug_assert_eq!(keys.count(), samples.len());
+        let mut gradient_samples = Vec::new();
+        for ([species_center], samples) in keys.iter_fixed_size().zip(samples) {
+            let builder = SamplesPerAtom {
+                species_center: SpeciesFilter::Single(species_center.i32())
+            };
+
+            gradient_samples.push(builder.gradients_for(systems, samples)?);
+        }
+
+        return Ok(gradient_samples);
+    }
+
+    fn components(&self, keys: &Labels) -> Vec<Vec<Arc<Labels>>> {
+        return vec![Vec::new(); keys.count()];
+    }
+
+    fn properties_names(&self) -> Vec<&str> {
+        return vec!["count"];
+    }
+
+    fn properties(&self, keys: &Labels) -> Vec<Arc<Labels>> {
+        let mut properties = LabelsBuilder::new(self.properties_names());
+        properties.add(&[0]);
+        let properties = Arc::new(properties.finish());
+
+        return vec![properties; keys.count()];
+    }
+
+    fn compute(&mut self, _: &mut [Box<dyn System>], descriptor: &mut TensorMap) -> Result<(), Error> {
+        assert_eq!(descriptor.keys().names(), ["species_center"]);
+
+        for (_, mut block) in descriptor.iter_mut() {
+            let values = block.values_mut();
+            let array = values.data.as_array_mut();
+            array.fill(1.0);
+
+            if let Some(gradient) = block.gradient_mut("positions") {
+                let array = gradient.data.as_array_mut();
+                array.fill(0.0);
+            } else if let Some(gradient) = block.gradient_mut("cell") {
+                let array = gradient.data.as_array_mut();
+                array.fill(0.0);
+            }
+        }
```

Comment:
Are these necessary? Aren't all Rust arrays already initialized (I assume to zero)?

Reply:
Yes, you are right. We can leave them as they are!